src/server/editorServices.ts

```diff
@@ -1492,47 +1718,111 @@ namespace ts.server {
+                ([path, isRootOfInferredProject]) => {
+                    if (!ignoreIfNotRootOfInferredProject || isRootOfInferredProject) {
+                        const info = this.getScriptInfoForPath(path as Path);
+                        Debug.assert(!!info);
+                        return info;
+                    }
+                }
+            );
+            this.reloadConfiguredsProjectForFiles(openFiles, /*delayReload*/ true);
+            this.delayInferredProjectsRefresh();
+        }
+
+        /**
+         * This function goes through all the openFiles and tries to file the config file for them.
+         * If the config file is found and it refers to existing project, it reloads it either immediately
+         * or schedules it for reload depending on delayReload option
+         * If the there is no existing project it just opens the configured project for the config file
+         */
+        private reloadConfiguredsProjectForFiles(openFiles: ScriptInfo[], delayReload: boolean) {
+            const updatedProjects = createMap<true>();
             // try to reload config file for all open files
-            for (const info of this.openFiles) {
-                this.openOrUpdateConfiguredProjectForFile(info.fileName);
+            for (const info of openFiles) {
+                // This tries to search for a tsconfig.json for the given file. If we found it,
+                // we first detect if there is already a configured project created for it: if so,
+                // we re- read the tsconfig file content and update the project only if we havent already done so
+                // otherwise we create a new one.
+                const configFileName = this.getConfigFileNameForFile(info);
+                if (configFileName) {
+                    const project = this.findConfiguredProjectByProjectName(configFileName);
+                    if (!project) {
+                        this.createConfiguredProject(configFileName, info.fileName);
+                        updatedProjects.set(configFileName, true);
+                    }
+                    else if (!updatedProjects.has(configFileName)) {
+                        if (delayReload) {
+                            project.pendingReload = true;
+                            this.delayUpdateProjectGraph(project);
+                        }
+                        else {
+                            this.reloadConfiguredProject(project);
+                        }
+                        updatedProjects.set(configFileName, true);
+                    }
+                }
+            }
+        }
+
+        /**
+         * Remove the root of inferred project if script info is part of another project
+         */
+        private removeRootOfInferredProjectIfNowPartOfOtherProject(info: ScriptInfo) {
+            // If the script info is root of inferred project, it could only be first containing project

```

Comment:
I've read this and I still don't understand why only the first project needs removing